Update to MSGETACLSTATUS

begoldsm opened the pull request:
This maintains the previous functionality for PowerShell callers and also adds in a new optional property called "permission"

Commit:
Need to make contents required since this is a PUT.

Will add extensions that make it optional in C# and zero it out to an
empty stream.

update documentation to reflect param required

undo this temp change

Issue: https://github.com/Azure/autorest/issues/884 is tracking support
for this being an optional param properly and not resulting in a null
reference. So I will keep this spec as is.

temp commit to remove the required flag

We need a way to have template parameters not be required when they are
in the host. I will work with Amar for this next week.

Revert "temp commit to remove the required flag"

This reverts commit be8b7bf40e56c8316f423a871ccf030efe47a8a1.

Work around for issue: 903

explicitly calling out our chunked transfer APIs until issue:
https://github.com/Azure/autorest/issues/903 is resolved.

Add missing method to delete all secrets in db

In hyak, we could do this by simply excluding the parameter. That is not
swagger compliant, so we need a separate entry.

Test out composites.

Remove composite specs since they don't work

update the required piece.

Try nesting again.

Removing get/set from descriptions

latest generation tool automatically adds it.

Fixed to not remove "Gets " from method descript

Add the get back to globals.

make typefieldinfo an array.

Adding offset support for append operations

This is an optional value used for unique implementations that know
exactly where in the stream they want to begin appending data.

Add file already exists exception for webhdfs

Add BadOffsetException to swagger.

Fix issues with bad rebase

had to copy back from azure and make the minor edits to these two .json
files.

Update specs for ADLA hive properties

Finalization of hive properties is in progress.

Add new catalog API version

this includes the new credential management layer.

Move the get with other operations for credentials

Add new property for job statistics

finalizingTimeUtc which gives us information about when the job
finalization started.

Removing this update from master

We are controlling this new update in a separate branch that is going to
go live later.

Update specs for ADLA hive properties

Finalization of hive properties is in progress.

Add new catalog API version

this includes the new credential management layer.

Move the get with other operations for credentials

Add new property for job statistics

finalizingTimeUtc which gives us information about when the job
finalization started.

Removing this update from master

We are controlling this new update in a separate branch that is going to
go live later.

Update job.json

minor comment update to trigger python and node rebuild of SDK.

Update to use MSGetAclStatus for ADLS

This keeps the existing behavior, which we need for proper presentation
of ACLs in PowerShell.

Add permission property to the AclStatus object

The ACL status object now includes a permission property.
Sticky bit is not supported so should not be modified.

John-Hart commented on arm-datalake-store/filesystem/2015-10-01-preview/swagger/filesystem.json:
```diff
@@ -1161,8 +1161,14 @@
+          "type": "integer",
+		  "format": "int32",
+          "description": "The octal representation of the unnamed user, mask and other permissions."
+        },
```
can you fix the formatting here?

begoldsm committed:
ADLS Swagger updates

Adding support for:
expiry (getting and setting)
ACLs (return microsoft specific ACL information, and remove full acls
and default acls)

Missed file on previous commit.

begoldsm committed:
fix tabbing and remove unused subscriptionID

begoldsm commented:
@John-Hart good catch! Tabbing fixed back to 2 spaces (curse you notepad++!)

amarzavery commented on arm-datalake-store/filesystem/2015-10-01-preview/swagger/filesystem.json:
```diff
@@ -116,6 +116,72 @@
+        ],
+        "responses": {
+          "200": {
+            "description": ""
```
no empty descriptions allowed

amarzavery commented on arm-datalake-store/filesystem/2015-10-01-preview/swagger/filesystem.json:
```diff
@@ -872,7 +938,91 @@
+        ],
+        "responses": {
+          "200": {
+            "description": ""
```
no empty description

amarzavery commented on arm-datalake-store/filesystem/2015-10-01-preview/swagger/filesystem.json:
```diff
@@ -872,7 +938,91 @@
+        ],
+        "responses": {
+          "200": {
+            "description": ""
```
same as above

begoldsm commented:
@amarzavery I will put in descriptions for the new stuff in this PR, and then we will need to have another documentation specific PR later to get the rest of the specs to be fully documented.

amarzavery commented:
sure

begoldsm committed:
Update the descriptions.

begoldsm committed:
Enable Encryption support for ADLS

amarzavery commented on arm-datalake-store/account/2015-10-01-preview/swagger/account.json:
```diff
@@ -679,6 +715,69 @@
       },
       "description": "Data Lake Store firewall rule list information."
     },
+    "EncryptionIdentity": {
```
I dont see this model being referenced anywhere.

begoldsm replied:
It is being referenced in DataLakeStoreAccount

amarzavery replied:
sorry. I got it.

begoldsm commented on arm-datalake-store/account/2015-10-01-preview/swagger/account.json:
```diff
@@ -757,6 +886,10 @@
           "readOnly": true,
           "description": "the account subscription ID."
         },
+        "identity": {
```
Right here is where it is being reference @amarzavery. I just double checked that I didn't spelly it wrong.

AutorestCI commented:
No modification for Ruby

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-node.svg?branch=RestAPI-PR492)](https://travis-ci.org/AutorestCI/azure-sdk-for-node) https://github.com/Azure/azure-sdk-for-node/pull/1861

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-python.svg?branch=RestAPI-PR492)](https://travis-ci.org/AutorestCI/azure-sdk-for-python) https://github.com/Azure/azure-sdk-for-python/pull/760
